### PR TITLE
[17.0][MIG] stock_picking_analytic: Migration to 17.0 

### PR DESCRIPTION
**Pull request by peluko00**
Module migrated to version 17.0

cc https://github.com/APSL 159231
@miquelalzanillas @lbarry-apsl @javierobcn @mpascuall @BernatObrador @ppyczko please review

- [Issue] https://github.com/OCA/account-analytic/issues/607

Depends on:

- [ stock_analytic] https://github.com/OCA/account-analytic/pull/614

### CODE REVIEW

**Review by mpascuall — APPROVED** [review]
Working good, checked in runboat

**Comment by peluko00** [comment]
Hi @StefanRijnhart seems, it's ready to merge

**Comment by StefanRijnhart** [comment]
@peluko00 Can you rebase and drop the test requirement?

**Comment by peluko00** [comment]
> @peluko00 Can you rebase and drop the test requirement?

Done!

File: stock_picking_analytic/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        help="Technical field. Use to store the value of"
+        "analytic_distribution if there is no lines",
+    )
+    analytic_distribution = fields.Json(inverse="_inverse_analytic_distribution")
+    # Assign a dummy search arg to the field to avoid triggering the warning.
+    # Without this, use of move_ids_without_package in @api.depends() of the compute
+    # method would trigger a warning from
+    # https://github.com/odoo/odoo/blob/0f84366/odoo/fields.py#L808-L812
+    # pylint: disable=method-search
+    move_ids_without_package = fields.One2many(search="[]")
```
Comment (StefanRijnhart) [inline stock_picking_analytic/models/stock_picking.py]: The field is no longer a computed field but instead a One2many field with a domain. Maybe this party trick that served us well in the past can and should be removed in this version? See https://github.com/odoo/odoo/blob/3b42467/addons/stock/models/stock_picking.py#L450-L451

**Review by StefanRijnhart — CHANGES_REQUESTED** [review]

**Review by BernatObrador — APPROVED** [review]
LGTM, tested in runboat

File: stock_picking_analytic/models/stock_picking.py (hunk 1 above)
Comment (peluko00) [reply stock_picking_analytic/models/stock_picking.py]: Yees, is not longer to be use. Thanks for the review

**Review by StefanRijnhart — APPROVED** [review]
Alright! (code review)

**Review by ppyczko — APPROVED** [review]
LGTM, tested in runboat

**Comment by peluko00** [comment]
> Alright! (code review)

Seems like i'ts ready to merge

**Comment by StefanRijnhart** [comment]
@peluko00 I don't think any of the PSC members of this repo have chimed in yet, so let's give it at least the 5 days until after creation of the PR that is conventional. Oca-bot will notify us when it's time.

**Comment by peluko00** [comment]
@rousseldenis  can you review please

**Comment by StefanRijnhart** [comment]
@peluko00 please note that reviews and other involvement in OCA projects are on a voluntary basis and that pings are not always appreciated.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**